packages/utilities/psammead-rich-text-transforms/src/candy-xml.js

```diff
@@ -44,13 +44,25 @@ const createUrlLink = element => {
   return urlLink(text, locator, blocks);
 };
 
+const handleSupportedNodes = (childNode, attributes, acc) => {
+  const block = xmlNodeToBlock(childNode, attributes);
+  const blocks = Array.isArray(block) ? block : [block];
+  return [...acc, ...blocks];
+};
+
+const handleUnsupportedNodes = (childNode, attributes) =>
+  convertToBlocks(childNode, attributes);
+
 const convertToBlocks = (node, attributes = []) =>
   pathOr([], ['elements'], node).reduce((acc, childNode) => {
-    if (isXmlNodeSupported(childNode)) {
-      const block = xmlNodeToBlock(childNode, attributes);
-      const blocks = is(Array, block) ? block : [block];
-      return [...acc, ...blocks];
+    if (isXmlNodeSupported(childNode, attributes)) {
+      return handleSupportedNodes(childNode, attributes, acc);
     }
+
+    if (Array.isArray(childNode.elements)) {
```

Comment:
Can this not be an array? can it be an empty array?

Reply:
Not 100% sure what you are asking. Do you mean this could be `if (childNode.elements)` and we call `convertToBlocks()` if `elements` exists at all on `childNode`?

Reply:
SGTM